packages/web-runtime/src/services/auth/userManager.ts

```diff
@@ -0,0 +1,240 @@
+        }),
+
+        scope: 'profile',
+        loadUserInfo: false,
+        metadata: {
+          issuer: oAuth2.url,
+          authorization_endpoint: oAuth2.authUrl,
+          token_endpoint: oAuth2.url,
+          userinfo_endpoint: ''
+        }
+      })
+    }
+
+    Log.setLevel(Log.INFO)
+
+    super(openIdConfig)
+    this.storePrefix = storePrefix
+    this.clientService = options.clientService
+    this.configurationManager = options.configurationManager
+    this.store = options.store
+  }
+
+  /**
+   * Looks up the access token of an already loaded user without enforcing a signin if no user exists.
+   *
+   * @return (string|null)
+   */
+  async getAccessToken(): Promise<string | null> {
+    const user = await this.getUser()
+    return user?.access_token
+  }
+
+  async removeUser(unloadReason: UnloadReason = 'logout') {
+    this._unloadReason = unloadReason
+    await super.removeUser()
+  }
+
+  get unloadReason(): UnloadReason {
+    return this._unloadReason
+  }
+
+  getAndClearPostLoginRedirectUrl(): string {
+    const url = sessionStorage.getItem(postLoginRedirectUrlKey) || '/'
+    sessionStorage.removeItem(postLoginRedirectUrlKey)
+    return url
+  }
+
+  setPostLoginRedirectUrl(url?: string): void {
+    if (url) {
+      sessionStorage.setItem(postLoginRedirectUrlKey, url)
+    } else {
+      sessionStorage.removeItem(postLoginRedirectUrlKey)
+    }
+  }
+
+  updateContext(accessToken: string): Promise<void> {
+    const userKnown = !!this.store.getters.user.id
+    const accessTokenChanged = this.store.getters['runtime/auth/accessToken'] !== accessToken
+    if (!accessTokenChanged) {
+      return this.updateAccessTokenPromise
+    }
+    this.store.commit('runtime/auth/SET_ACCESS_TOKEN', accessToken)
+
+    this.updateAccessTokenPromise = (async () => {
+      this.initializeOwnCloudSdk(accessToken)
+
+      if (!userKnown) {
+        await this.fetchUserInfo(accessToken)
+        this.store.commit('runtime/auth/SET_USER_CONTEXT_READY', true)
+      }
+    })()
+    return this.updateAccessTokenPromise
+  }
+
+  private initializeOwnCloudSdk(accessToken): void {
+    const options = {
+      baseUrl: this.configurationManager.serverUrl,
+      auth: {
+        bearer: accessToken
+      },
+      headers: {
+        'X-Requested-With': 'XMLHttpRequest'
+      }
+    }
+    if (this.store.getters.user.id) {
+      ;(options as any).userInfo = {
+        id: this.store.getters.user.id,
+        'display-name': this.store.getters.user.displayname,
+        email: this.store.getters.user.email
+      }
+    }
+
+    this.clientService.owncloudSdk.init(options)
+  }
+
+  private async fetchUserInfo(accessToken): Promise<void> {
+    const login = await this.clientService.owncloudSdk.getCurrentUser()
+    await this.fetchCapabilities({ accessToken })
+
+    // FIXME: Can be removed as soon as the uuid is integrated in the OCS api
+    // see https://github.com/owncloud/ocis/issues/3271
```

Comment:
I think the comment moved to a weird place when we added the roles here. I think it originally referred to https://github.com/owncloud/web/pull/7072/files#diff-7c13ed3f4e1b32127b202bef84f63748484b2cd21a50379a369a7c179137d149R206

We should probably elaborate a bit further here now...